server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -0,0 +1,321 @@
+                    // TODO - should we put the weights back to zone, since we weighed away the zone before we started registering the metadata
+                    // TODO - should we modify logic of logging for ease of debugging?
+                    if (e instanceof DecommissionFailedException) {
+                        logger.error(() -> new ParameterizedMessage("failed to decommission attribute [{}]", decommissionAttribute.toString()), e);
+                    } else if (e instanceof NotClusterManagerException) {
+                        logger.info(() -> new ParameterizedMessage(
+                            "cluster-manager updated while executing request for decommission attribute [{}]",
+                            decommissionAttribute.toString()), e);
+                    } else {
+                        // could be due to on longer cluster manager
+                        clusterService.submitStateUpdateTask(
+                            "decommission_failed",
+                            new ClusterStateUpdateTask(Priority.URGENT) {
+                                @Override
+                                public ClusterState execute(ClusterState currentState) throws Exception {
+                                    Metadata metadata = currentState.metadata();
+                                    Metadata.Builder mdBuilder = Metadata.builder(metadata);
+                                    logger.info("decommission request for attribute [{}] failed", decommissionAttribute.toString());
+                                    DecommissionAttributeMetadata decommissionAttributeMetadata = new DecommissionAttributeMetadata(
+                                        decommissionAttribute,
+                                        DecommissionStatus.DECOMMISSION_FAILED
+                                    );
+                                    mdBuilder.putCustom(DecommissionAttributeMetadata.TYPE, decommissionAttributeMetadata);
+                                    return ClusterState.builder(currentState).metadata(mdBuilder).build();
+                                }
+
+                                @Override
+                                public void onFailure(String source, Exception e) {
+                                    logger.error(() -> new ParameterizedMessage(
+                                        "failed to mark status as DECOMMISSION_FAILED for decommission attribute [{}]",
+                                        decommissionAttribute.toString()), e);
+//                                    listener.onFailure(e);
+                                }
+                            }
+                        );
+                    }
+                    listener.onFailure(e);
+                }
+
+                @Override
+                public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
+                    if (!newState.equals(oldState)) {
```

Comment:
instead of comparing states, we could verify that the new state is holding the expected decommission attribute?

Reply:
Makes sense. Updating it